### PR TITLE
Edit target view

### PR DESCRIPTION
**Pull request by smanini1**
#### Description:

Added didSelect func on homeController to handle annotation selections
Added necessary logic to handle Target selection on homeController
Added necessary logic to handle Target selection on TargetController

---

#### Preview:

ezgif com-video-to-gif (8)

### CODE REVIEW

File: ios-target/View/Home/HomeViewController.swift [hunk 1]
```diff
@@ -36,13 +36,26 @@ class HomeViewController: UIViewController {
...
       let viewController = segue.destination as? TargetFormViewController {
-      viewController.delegate = self
-      viewController.viewModel.topics = viewModel.topics
-      viewController.viewModel.targetLocation.longitude = viewModel.locationCoordinates.longitude
-      viewController.viewModel.targetLocation.latitude = viewModel.locationCoordinates.latitude
+      setupController(viewController)
     }
   }
   
+  func setupController(_ viewController: TargetFormViewController) {
```
Comment (germanStabile) [inline ios-target/View/Home/HomeViewController.swift:43]: be more explicit on the naming `setupTargetController`

File: ios-target/View/Home/TargetFormViewController.swift [hunk 2]
```diff
@@ -25,7 +26,20 @@ class TargetFormViewController: ModalViewController {
...
     selectTopicButton.addBorder(color: .black, weight: 1)
+    viewModel.setTarget()
+    setView()
+  }
+  
+  func setView() {
+    titleTextField.text = viewModel.targetTitle
+    areaTextField.text = viewModel.targetArea != 0 ? String(viewModel.targetArea) : ""
+    guard viewModel.targetTopic != 0,
+    let index = viewModel.topics.index(where: {$0.id == viewModel.targetTopic}) else { return }
```
Comment (germanStabile) [inline ios-target/View/Home/TargetFormViewController.swift:39]: indentation looks crooked in this guard.

```swift 

guard
   viewModel.targetTopic != 0,
   let index = viewModel.topics.index(where: {$0.id == viewModel.targetTopic})
else { return }

File: ios-target/View/Home/TargetFormViewController.swift [hunk 3]
```diff
@@ -25,7 +26,20 @@ class TargetFormViewController: ModalViewController {
     super.viewDidLoad()
     viewModel.delegate = self
     createTargetButton.enable(false)
+    deleteTargetButton.isHidden = viewModel.isEditingTarget
     selectTopicButton.addBorder(color: .black, weight: 1)
+    viewModel.setTarget()
+    setView()
+  }
+  
+  func setView() {
```
Comment (germanStabile) [inline ios-target/View/Home/TargetFormViewController.swift:35]: more like `setupViews` or `configureViews`

**Review by germanStabile — APPROVED** [review]
LG!

File: ios-target/View/Home/TargetFormViewController.swift (hunk 2 above)
Comment (mcousillas6) [inline ios-target/View/Home/TargetFormViewController.swift:39]: missing indentation here

**Review by mcousillas6 — APPROVED** [review]

**Pull request merged**